includes/event/event-repository.php

```diff
@@ -454,10 +476,25 @@ private function execute_events_query( int $page, int $page_size, array $args, a
 			$args['post__in'] = $filter_by_ids;
 		}
 
-		$query  = new WP_Query( $args );
-		$posts  = $query->get_posts();
-		$events = array();
+		if ( null !== $user_id ) {
+			$user_id_filter_callback = 'Wporg\TranslationEvents\Event\add_user_id_where_clause_to_events_query';
+			// Only return events for which this user is an attendee, or (optionally) the event author.
+			// We use a filter to modify the where clause of the query.
+			// The filter removes itself, so it will only apply to the next query.
+			add_filter( 'posts_where', $user_id_filter_callback, 10, 2 );
+			$args['translation_events_user_id']                 = $user_id;
+			$args['translation_events_include_created_by_user'] = $include_created_by_user;
+		}
 
+		$query = new WP_Query( $args );
+		$posts = $query->get_posts();
+
+		if ( isset( $user_id_filter_callback ) ) {
+			// Remove the filter, so it only applies to this query.
+			remove_filter( 'posts_where', $user_id_filter_callback );
```

Comment:
Note this will work because we use the default priority 10, otherwise we'd need to specify it.

Reply:
Good point. I've now added a variable so it will always be set to the same value as the `add_filter()`.